src/ripple/shamap/SHAMap.h

```diff
@@ -149,27 +151,31 @@ class SHAMap
             Can throw SHAMapMissingNode
         @note This can cause NodeStore reads
     */
-    // VFALCO NOTE This should return `const&` when SHAMapTreeNode
-    //             stores std::shared_ptr<SHAMapItem const>
-    std::shared_ptr<SHAMapItem const>
-    fetch (uint256 const& key) const;
+    std::shared_ptr<SHAMapItem const> const&
+        fetch (uint256 const& key) const;
 
     // VFALCO NOTE Is "save a copy" the in imperative or indicative mood?
     // save a copy if you only need a temporary
-    std::shared_ptr<SHAMapItem> peekItem (uint256 const& id) const;
-    std::shared_ptr<SHAMapItem> peekItem (uint256 const& id, uint256 & hash) const;
-    std::shared_ptr<SHAMapItem> peekItem (uint256 const& id, SHAMapTreeNode::TNType & type) const;
+    std::shared_ptr<SHAMapItem const> const& peekItem (uint256 const& id) const;
```

Comment:
I'll address the comment.

In general, the decision of whether a copy is needed is the client's to make.  The server (SHAMap) isn't going to make a copy.  This is a change in API that consistently puts the decision in one place (the client), so that future changes can more easily decide if it is safe to eliminate a copy entirely (the one on the client side), and can do so on  a case by case basis.